### ISSUE TITLE
IBB file transfer fails on the second file due to an issue with Eventmachine

### ISSUE THREAD

**Issue by fyafighter** [adhearsion/blather#61]
There is a problem with IBB file transfers where the second file raises a broken pipe error. This is due to the following issue:

https://github.com/eventmachine/eventmachine/issues/200

Ultimately the problem is the attach method in eventmachine is causing some drama. As far as I can tell the best fix (without an EM solution) will be to change the file saving method to be a thread on the EM pool that is reading the chunks from a queue and writing them to disk. This should allow it to work and also not block the reactor thread, but it less than ideal.

We will be including this solution in our project to get unblocked (overriding the IBB class). I would like to know if this would be something you are interested in for the blather gem, or would you rather wait for the EM solution. Let me know, and I would be happy to generate another pull request.
